File: Ander systeem/Core.ApplicationServices/DTO/DataConfig/DataRequestDTO.cs
namespace Leap.ApplicationServices.DTO.DataConfig
{
    public class DataRequestDTO
    {
        // Data request class used to be send to the Python service.
        public string WorkspaceId { get; set; } = string.Empty;
        public string Token { get; set; } = string.Empty;
        public string DataSource { get; set; } = string.Empty;
        public long StartDateUnix { get; set; }
        public long EndDateUnix { get; set; }
        public string Project { get; set; } = string.Empty;
        public int Points { get; set; } = 5;
        public int? Timelevel { get; set; }
        public float? TimelevelRange { get; set; }
        public IEnumerable<int> ObservationIds { get; set; } = [];
        public IEnumerable<int> ValueTypeIds { get; set; } = [];
        public bool ProvideData { get; set; } = false;
    }
}

File: App Python/Domain/DTO/DataRequestDTO.py
from typing import Optional

from pydantic import BaseModel


class DataRequestDTO(BaseModel):
    """
    This is the request body needed to retrieve column and preview data.
    The assumption is that an data source is already configured.
    """

    WorkspaceId: str
    Token: str
    DataSource: str
    StartDateUnix: int
    EndDateUnix: int
    Project: str
    Points: int = 20
    Timelevel: Optional[int] = -1
    TimelevelRange: Optional[float] = -1.0
    ObservationIds: list[int]
    ValueTypeIds: list[int]
    ProvideData: bool
